### PR TITLE
ENG-14120: also catch SQLExeception on the loadTable path

### PR DESCRIPTION
**Pull request by JadenW**
Has to propagate the sqlException through the processLoadTable() method, since the deleteTupleStorage is only implemented in persistentTable.

### CODE REVIEW

**Comment by rmorgenstein** [comment]
Please add a test for this.

File: src/ee/execution/VoltDBEngine.cpp [hunk 1]
```diff
@@ -1586,7 +1586,7 @@ VoltDBEngine::loadTable(int32_t tableId,
     }
 
     // When loading a replicated table, behavior should be:
-    //   ConstraintFailureExceptions may be thrown on the lowest site thread.
+    //   SQLException may be thrown on the lowest site thread.
```
Comment (yzhang1991) [inline src/ee/execution/VoltDBEngine.cpp]: Revert this change please. This paragraph still describes the behavior of ConstraintFailureExceptions.

File: src/ee/execution/VoltDBEngine.cpp [hunk 2]
```diff
@@ -1586,7 +1586,7 @@ VoltDBEngine::loadTable(int32_t tableId,
...
     //   If throwUniqueViolations is true:
```
Comment (yzhang1991) [inline src/ee/execution/VoltDBEngine.cpp]: Replace `throwUniqueViolations` with `returnConflictRows`, keep it consistent with the function signature.
There is another occurrence of `throwUniqueViolations` in line 1593, please replace it as well.

File: src/ee/execution/VoltDBEngine.cpp [hunk 3]
```diff
@@ -1586,7 +1586,7 @@ VoltDBEngine::loadTable(int32_t tableId,
...
     //       Lowest site thread: throw the exception.
     //       Other site threads: throw replicated table exceptions.
```
Comment (yzhang1991) [inline src/ee/execution/VoltDBEngine.cpp]: In line 1597:
>  For all other kinds of exceptions, throw a FatalException...

This needs to be updated.
No FatalExceptions for SQLExceptions, we will always rethrow it.

File: src/ee/execution/VoltDBEngine.cpp [hunk 4]
```diff
@@ -1649,7 +1653,22 @@ VoltDBEngine::loadTable(int32_t tableId,
         if (!returnConflictRows) {
             std::ostringstream oss;
             oss << "Replicated load table failed (constraint violation) on other thread for table \""
-                << table->name() << "\".";
+                << table->name() << "\".\n";
+            VOLT_DEBUG("%s", oss.str().c_str());
+            throw SerializableEEException(VOLT_EE_EXCEPTION_TYPE_REPLICATED_TABLE, oss.str().c_str());
+        }
+        else {
```
Comment (yzhang1991) [inline src/ee/execution/VoltDBEngine.cpp]: You don't need an `else` here, right?

File: src/ee/execution/VoltDBEngine.cpp [hunk 5]
```diff
@@ -1649,7 +1653,22 @@ VoltDBEngine::loadTable(int32_t tableId,
...
+    else if (s_loadTableException == VOLT_EE_EXCEPTION_TYPE_SQL) {
+        // An constraint failure exception was thrown on the lowest site thread and
+        // handle it on the other threads too.
+        if (!returnConflictRows) {
+            std::ostringstream oss;
+            oss << "Replicated load table failed (sql exception) on other thread for table \""
+                << table->name() << "\".\n";
             VOLT_DEBUG("%s", oss.str().c_str());
             throw SerializableEEException(VOLT_EE_EXCEPTION_TYPE_REPLICATED_TABLE, oss.str().c_str());
         }
```
Comment (yzhang1991) [inline src/ee/execution/VoltDBEngine.cpp]: Same here, I think we do not need an `else` in line 1675?

File: src/ee/storage/persistenttable.cpp [hunk 6]
```diff
@@ -1571,6 +1571,98 @@ std::string PersistentTable::debug(const std::string& spacer) const {
     return buffer.str();
 }
 
+/**
+     * Loads tuple data from the serialized table.
```
Comment (yzhang1991) [inline src/ee/storage/persistenttable.cpp]: Fix the indentation for the comments here, please.

File: src/ee/storage/persistenttable.cpp [hunk 7]
```diff
@@ -1571,6 +1571,98 @@ std::string PersistentTable::debug(const std::string& spacer) const {
...
+     * Used for snapshot restore and bulkLoad
+     */
+void PersistentTable::loadTuplesForLoadTable(SerializeInputBE& serialInput,
+                                      Pool* stringPool,
```
Comment (yzhang1991) [inline src/ee/storage/persistenttable.cpp]: This entire function was not indented correctly, please fix it.

File: tests/frontend/org/voltdb/regressionsuites/TestLoadingSuite.java [hunk 8]
```diff
@@ -143,6 +143,7 @@ public void testMultiPartitionLoad() throws Exception {
             fail();
         }
         catch (ProcCallException e) {
+            e.printStackTrace();
```
Comment (yzhang1991) [inline tests/frontend/org/voltdb/regressionsuites/TestLoadingSuite.java]: We do not need to print out the stack trace here because the exception was expected?

File: tests/frontend/org/voltdb/regressionsuites/TestLoadingSuite.java [hunk 9]
```diff
@@ -154,7 +155,21 @@ public void testMultiPartitionLoad() throws Exception {
             fail(); // prev stmt should throw exception
         }
         catch (ProcCallException e) {
+            e.printStackTrace();
```
Comment (yzhang1991) [inline tests/frontend/org/voltdb/regressionsuites/TestLoadingSuite.java]: No need to print the stack trace.

File: tests/frontend/org/voltdb/regressionsuites/TestLoadingSuite.java [hunk 10]
```diff
@@ -154,7 +155,21 @@ public void testMultiPartitionLoad() throws Exception {
...
+        // test rollback to a replicated table (sql exception)
+        table = m_template.clone(100);
+        table.addRow(6, 1, 5, "6", 5.0);
+        table.addRow(7, 1, 5, "777", 5.0);
+        try {
+            r = client.callProcedure("@LoadMultipartitionTable", "REPLICATED", upsertMode, table);
+            fail(); // prev stmt should throw exception
+        }
+        catch (ProcCallException e) {
+            e.printStackTrace();
```
Comment (yzhang1991) [inline tests/frontend/org/voltdb/regressionsuites/TestLoadingSuite.java]: No need to print the stack trace.
